### PR TITLE
Simpler tarball preparation

### PR DESCRIPTION
**Pull request by mvidner**
https://trello.com/c/kYLRhKDf/319-do-not-require-all-dependencies-to-build-the-libyui-tarballs-at-jenkins

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
-Packaging::Tasks.load_tasks(:exclude => ["tarball.rake"])
-
-BUILDDIR = "build"
+require "packaging"
+
+# @param s   [String] '... SET( VERSION_MAJOR "3") ...'
+# @param key [String] 'VERSION_MAJOR'
+# @return "3"
+def cmake_value(s, key)
+ m = /SET\s*\(\s*#{key}\s+"([^"]*)"\s*\)/.match(s)
```
Comment (jreidinger) [inline Rakefile]: better `Regexp.escape(key)` unless you really want to support regexp

File: package/libyui.spec [hunk 2]
```diff
@@ -16,36 +16,48 @@
...
+This is the user interface engine that provides the abstraction from
+graphical user interfaces (Qt, Gtk) and text based user interfaces
+(ncurses).
 
-%package -n @PROJECTNAME@@SONAME_MAJOR@
+Originally developed for YaST, it can now be used independently of
+YaST for generic (C++) applications. This package has very few
+dependencies.
+
+%package -n libyui7
```
Comment (jreidinger) [inline package/libyui.spec]: I think this 7 should be in some macro, as it is spread over spec file, so change one value change all occurrences.

**Comment by mvidner** [comment]
What is left now:
- "DRY spec files with defines" in all the other PRs
- bump versions and add .changes entries to all PRs: 

> - Reorganized git for easier tarball creation:
  - RPM spec files are kept in git verbatim, not as templates
  - no longer call PREP_SPEC_FILES in CMakeLists.common
- No functional change but version bumped to push the package down the pipeline

**Comment by mvidner** [comment]
@jreidinger @nilxam I think these are all ready now, please review.

**Comment by mvidner** [comment]
To test these you will need to refer to the new version of libyui.rpm as the dependent packages still use its buildtools and with the old version they would try to touch their *.spec.cmake templates which are no longer there. So use:
```sh
rake -f ../libyui/Rakefile osc:build"[-k /tmp -p /tmp]"
```

**Comment by jreidinger** [comment]
code change is OK. What I still do not like is refering to rakefile from different project, why not simple create shared libyui-rake-tasks? or simple copy it as now it is much simplier.

**Comment by mvidner** [comment]
Fair point about the Rakefile. I am reluctant to change it in this PR, since it implies touching all the Jenkins jobs (something I managed to avoid so far), and brings questions about priorities of further cleanups of the build system.
I would prefer to leave the Rakefile as is for now.

**Pull request merged**